client.go

```diff
@@ -474,27 +477,25 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		// Check if we should continue with retries.
 		checkOK, checkErr := c.CheckRetry(req.Context(), resp, err)
 
-		if logger != nil {
-			if err != nil {
+		if err != nil {
+			switch v := logger.(type) {
+			case Logger:
+				v.Printf("[ERR] %s %s request failed: %v", req.Method, req.URL, err)
```

Comment:
This line wasn't removed like the unified diff makes it appear, if you view this PR in Split mode it becomes clearer I just moved the blocks around when I removed the `if logger != nil! check.